resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -176,16 +179,6 @@ private[spark] abstract class YarnSchedulerBackend(
   }
 
   /**
-   * Reset the state of SchedulerBackend to the initial state. This is happened when AM is failed
-   * and re-registered itself to driver after a failure. The stale state in driver should be
-   * cleaned.
-   */
-  override protected def reset(): Unit = {
```

Comment:
OK, then if `ExecutorAllocationManager#reset` is not required, then why would you still keep that method?

Also should we clean this two fields during AM restart in `ExecutorAllocationManager`?

```
    executorsPendingToRemove.clear()
    removeTimes.clear()
```